lib/scoped/index.js

```diff
@@ -5,16 +5,18 @@ var EventEmitter = require('events').EventEmitter
 function scoped (state, api, type) {
   var emitter = new EventEmitter()
 
+  var boundTypeChecker = require('../utils/type-check-wrapper').bind(null, type)
+
   var scopedApi = {
-    add: require('./add').bind(null, type, api),
-    find: require('./find').bind(null, type, api),
-    findOrAdd: require('./find-or-add').bind(null, type, api),
-    findAll: require('./find-all').bind(null, type, api),
-    update: require('./update').bind(null, type, api),
-    updateOrAdd: require('./update-or-add').bind(null, type, api),
-    updateAll: require('./update-all').bind(null, type, api),
-    remove: require('./remove').bind(null, type, api),
-    removeAll: require('./remove-all').bind(null, type, api),
+    add: boundTypeChecker(require('./add').bind(null, api)),
+    find: boundTypeChecker(require('./find').bind(null, api)),
+    findOrAdd: boundTypeChecker(require('./find-or-add').bind(null, api)),
+    findAll: require('./find-all').bind(null, api, type),
+    update: boundTypeChecker(require('./update').bind(null, api)),
+    updateOrAdd: boundTypeChecker(require('./update-or-add').bind(null, api)),
+    updateAll: boundTypeChecker(require('./update-all').bind(null, api)),
+    remove: boundTypeChecker(require('./remove').bind(null, api)),
+    removeAll: boundTypeChecker(require('./remove-all').bind(null, api)),
```

Comment:
I’m not sure about this implementation of checking the type. It’s good because it does not repeat code, but I wonder if it’s worth the mental overhead, compared to the alternative to put a type-check in every of these methods? This code is pretty hard to follow, we are binding a function that we pass to a pre-bound function that returns a new function ... hmmm 🤔